consensus/src/main/kotlin/maru/consensus/qbft/QbftBlockCreatorFactory.kt

```diff
@@ -50,15 +50,20 @@ class QbftBlockCreatorFactory(
         beaconChain = beaconChain,
         round = round,
       )
-    return EagerQbftBlockCreator(
-      manager = manager,
-      delegate = delayedQbftBlockCreator,
-      finalizationStateProvider = finalizationStateProvider,
-      blockBuilderIdentity = blockBuilderIdentity,
-      beaconChain = beaconChain,
-      nextBlockTimestampProvider = nextBlockTimestampProvider,
-      config = eagerQbftBlockCreatorConfig,
-      clock = clock,
-    )
+    val number = beaconChain.getLatestBeaconState().latestBeaconBlockHeader.number + 1u
+    return if (round == 0 && manager.hasStartedBlockBuilding()) {
```

Comment:
Alternative idea is to create `EagerQbftBlockCreator` only on the first call of this class without the use of `hasStartedBlockBuilding`. I'm not 100% sure if it will pass the tests though. Related to [this](https://github.com/Consensys/maru/pull/100/files#r2101917455) comment

Reply:
I like that idea, going to give it ago. If it works out it will be simpler and I don't need to change the `ExecutionLayerManager` then too.